src/ast/ast_build_reference_map.c

```diff
@@ -17,13 +22,13 @@ static void _AST_MapExpression(AST *ast, const cypher_astnode_t *exp) {
 	// In case of identifier.
 	if(type == CYPHER_AST_IDENTIFIER) {
 		const char *identifier_name = cypher_ast_identifier_get_name(exp);
-		_AST_UpdateRefMap(ast, identifier_name);
+		_AST_UpdateRefMap(ast, identifier_name, REFERENCED_ENTITY_HAS_IDENTIFIER);
 	} else if(type == CYPHER_AST_PROPERTY_OPERATOR) {
 		// In case of property.
 		exp = cypher_ast_property_operator_get_expression(exp);
 		assert(cypher_astnode_type(exp) == CYPHER_AST_IDENTIFIER);
 		const char *identifier_name = cypher_ast_identifier_get_name(exp);
-		_AST_UpdateRefMap(ast, identifier_name);
+		_AST_UpdateRefMap(ast, identifier_name, REFERENCED_ENTITY_IS_PROPERTY_VALUE);
```

Comment:
please rename `REFERENCED_ENTITY_IS_PROPERTY_VALUE` to `REFERENCED_ENTITY_HAS_PROPERTY`